integrations/sonarqube/client.py

```diff
@@ -284,82 +267,77 @@ async def get_single_project(self, project: dict[str, Any]) -> dict[str, Any]:
 
         return project
 
-    async def get_all_projects(self) -> AsyncGenerator[list[dict[str, Any]], None]:
-        """
-        Retrieve all projects from SonarQube API.
+    @cache_iterator_result()
+    async def get_projects(
+        self, params: dict[str, Any] = {}, enrich_project: bool = True
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        if self.organization_id:
+            params["organization"] = self.organization_id
+
+        async for projects in self._send_paginated_request(
+            endpoint=Endpoints.PROJECTS,
+            data_key="components",
+            method="GET",
+            query_params=params,
+        ):
+            if enrich_project:
```

Comment:
also I prefer that the default will be `false` and that calls to that method that need enriching will specify it explicitly with why